lib/player.js

```diff
@@ -3796,48 +3797,19 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
    * @param {string=} mimeType
    * @param {string=} codec
    * @param {string=} label
-   * @return {shaka.extern.Track}
+   * @return {!Promise.<shaka.extern.Track>}
```

Comment:
This just occurred to me as I was writing the commit message to merge your PR:

Changing the return type to a Promise is technically a breaking change.  Apps that use the return from `addTextTrack()` will break if they don't update to `await` the value or call `.then()`.  With strict semantic versioning, this means we can't put this PR into a release version until v4.0.0, which we currently have no timeline for.

Would you perhaps like to split it into two parts?  The rest of this change could be released in v3.1, while the Promise return value and async HEAD request for the MIME type could stay in the `master` branch until v4.

Reply:
I'm going to create a PR with just changing the error codes. But the HEAD request and the conversion from SRT to WebVTT would be left out.

@joeyparrish , Could you accept this change in a new method called `addAsyncTextTrack`? This would avoid problems.

Reply:
Absolutely.  We can also mark addTextTrack as deprecated and marked for removal whenever we decide to move to v4.

Reply:
For an example, see here: https://github.com/google/shaka-player/blob/50010869010c9caeadf5ec53de4c341a9f390cf1/lib/player.js#L1773-L1778

You could add a similar call to the top of addTextTrack to point people toward addTextTrackAsync.